wrapped_test.go

```diff
@@ -0,0 +1,248 @@
+package rules
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLeft(t *testing.T) {
+	boardState := &BoardState{
+		Width:  11,
+		Height: 11,
+		Snakes: []Snake{
+			{ID: "bottomLeft", Health: 10, Body: []Point{{0, 0}}},
+			{ID: "bottomRight", Health: 10, Body: []Point{{10, 0}}},
+			{ID: "topLeft", Health: 10, Body: []Point{{0, 10}}},
+			{ID: "topRight", Health: 10, Body: []Point{{10, 10}}},
+		},
+	}
+
+	snakeMoves := []SnakeMove{
+		{ID: "bottomLeft", Move: "left"},
+		{ID: "bottomRight", Move: "left"},
+		{ID: "topLeft", Move: "left"},
+		{ID: "topRight", Move: "left"},
+	}
+
+	r := WrappedRuleset{}
+
+	nextBoardState, err := r.CreateNextBoardState(boardState, snakeMoves)
+	require.NoError(t, err)
+	require.Equal(t, len(boardState.Snakes), len(nextBoardState.Snakes))
+
+	expectedSnakes := []Snake{
+		{ID: "bottomLeft", Health: 10, Body: []Point{{10, 0}}},
+		{ID: "bottomRight", Health: 10, Body: []Point{{9, 0}}},
+		{ID: "topLeft", Health: 10, Body: []Point{{10, 10}}},
+		{ID: "topRight", Health: 10, Body: []Point{{9, 10}}},
+	}
+	for i, snake := range nextBoardState.Snakes {
+		require.Equal(t, expectedSnakes[i].ID, snake.ID, snake.ID)
+		require.Equal(t, expectedSnakes[i].EliminatedCause, snake.EliminatedCause, snake.ID)
+		require.Equal(t, expectedSnakes[i].EliminatedBy, snake.EliminatedBy, snake.ID)
+		require.Equal(t, expectedSnakes[i].Body, snake.Body, snake.ID)
+	}
+}
```

Comment:
What's the thinking behind breaking these up into left/right/up/down? This could all be one big test with a single table of conditions, right?

Reply:
concurrency in tests. Grid tests are great (and would be more readable in this case I think), but this way the test runner can distribute the 4 different tests.

Reply:
Fair enough - I think the tests are fast enough at this stage that it's not worth micro-optimizing them, but I'm not too worried either way.